### PR TITLE
Fixes cached examples

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes: #1809

### CODE REVIEW

File: gradio/examples.py [hunk 1]
```diff
@@ -153,9 +153,9 @@ def __init__(
             self.cache_interface_examples()
 
         def load_example(example_id):
-            processed_example = self.processed_examples[example_id]
+            processed_input = self.processed_examples[example_id]
             if cache_examples:
-                processed_example += self.load_from_cache(example_id)
```
Comment (abidlabs) [inline gradio/examples.py:158]: The cached output was being appended again and again to the input, causing the bug

File: gradio/flagging.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def replace_flag_at_index(file_content):
         if self.encryption_key:
             output = io.StringIO()
             if not is_new:
-                with open(log_filepath, "rb") as csvfile:
+                with open(log_filepath, "rb", encoding="utf-8") as csvfile:
```
Comment (abidlabs) [inline gradio/flagging.py:166]: Separate problem (on Windows) was that the special characters in Turkish were not being encoded correctly

**Review by aliabid94 — CHANGES_REQUESTED** [review]
I got this error when running the demo/calculator/run.py and clicking on any example:
```
File "/Users/aliabid/projects/gradio/gradio/examples.py", line 159, in load_example
    if len(processed_example) == 1:
UnboundLocalError: local variable 'processed_example' referenced before assignment
```

**Review by freddyaboulton — COMMENTED** [review]
Thanks @abidlabs ! Looks good to me apart from the bug pointed out by @aliabid94 . 

Once that's fixed I think this is good to merge. I tested `image_classifier`, `gpt_j`, and the original space mentioned in the issue and they work with `cache_examples=True`

**Comment by abidlabs** [comment]
Yikes good catch @aliabid94. Will fix and retest

**Comment by abidlabs** [comment]
Ok fixed the issue @aliabid94 and @freddyaboulton. If you get a chance to make another pass, would be great. Otherwise, will go ahead and merge in shortly

**Review by freddyaboulton — APPROVED** [review]
Thank you @abidlabs !

**Comment by abidlabs** [comment]
Thanks @freddyaboulton!

**Pull request merged**